src/Symfony/Component/Process/ProcessUtils.php

```diff
@@ -93,8 +93,14 @@ public static function validateInput($caller, $input)
             if (is_scalar($input)) {
                 return (string) $input;
             }
+            while ($input instanceof \IteratorAggregate) {
+                $input = $input->getIterator();
+            }
+            if ($input instanceof \Iterator) {
+                return $input;
+            }
 
-            throw new InvalidArgumentException(sprintf('%s only accepts strings or stream resources.', $caller));
+            throw new InvalidArgumentException(sprintf('%s only accepts strings, Traversable objects or stream resources.', $caller));
```

Comment:
technically, your code won't accept Traversable implemented in C without implementing Iterator, or implemented in PHP with IteratorAggregate returning such a traversable.

you should use ``IteratorIterator`` instead to wrap any Traversable into an Iterator.